README.md

```diff
@@ -42,16 +42,16 @@ async ()=> {
 
 #### Transpilation / Polyfill
 
-There is no way of transpilating the actual effect of this proposal. But instead we can write a polyfill that keeps API consistent while it's using the main thread for parsing and stringifying tasks.
+There is no way of transpilating the actual effect of this proposal. But instead we can write a polyfill that keeps API consistent while it's using the main thread for parsing and stringifying tasks. We are using promises to perform the sync operation in the next processor tick.
 
 
 ```js
-JSON.stringifyAsync = JSON.stringifyAsync || function (object) {
-  return new Promise(resolve => resolve(JSON.stringify(object)));
+JSON.stringifyAsync = JSON.stringifyAsync || function stringifyAsync(...args) {
+ return Promise.resolve().then(()=> JSON.parse(...args));
 }
 
-JSON.parseAsync = JSON.parseAsync || function (string) {
-  return new Promise(resolve => resolve(JSON.parse(string)));
+JSON.parseAsync = JSON.parseAsync || function parseAsync(...args) {
```

Comment:
One problem with this approach - the length property of the function needs to match the number of required arguments, consistent with the rest of the spec. In other words, you'll want the async function polyfill's length to match that of the builtin sync function.